Libraries/LibWeb/Fetch/Response.cpp

```diff
@@ -85,15 +85,21 @@ GC::Ref<Response> Response::create(JS::Realm& realm, GC::Ref<Infrastructure::Res
 }
 
 // https://httpwg.org/specs/rfc9112.html#status.line
-static bool is_valid_status_text(StringView status_text)
+static bool is_valid_status_text(String const& status_text)
 {
-    // A status text is a valid status text if it matches the reason-phrase token production.
+    // A status text is a valid if it is either the empty string or matches the reason-phrase token production.
     // reason-phrase  = 1*( HTAB / SP / VCHAR / obs-text )
     // VCHAR          = %x21-7E
     // obs-text       = %x80-FF
-    return all_of(status_text, [](auto c) {
```

Comment:
Instead of removing the use of `all_of`, I think it would be nicer to continue using it, and just say: `all_of(status_text.code_points(), ...`

Though - it does seem wrong to consider the empty string as a valid reason phrase - that doesn't match the wording of the spec. So I think it would be best to follow the spec and move the `is_empty` check to the spec step that says:

>     // 2. If init["statusText"] is not the empty string and does not match the reason-phrase token production, then throw a TypeError.

Reply:
> Or maybe, we just remove the is_empty() check here since it is redundant
Thanks for the review. I took this approach. Please take another look.